src/components/AttachmentModal.js

```diff
@@ -74,22 +76,51 @@ class AttachmentModal extends PureComponent {
 
         this.state = {
             isModalOpen: false,
+            isConfirmModalOpen: false,
+            reportId: null,
+            showArrows: false,
+            file: {name: lodashGet(props, 'originalFileName', '')},
             isAttachmentInvalid: false,
             attachmentInvalidReasonTitle: null,
             attachmentInvalidReason: null,
-            file: null,
-            sourceURL: props.sourceURL,
             modalType: CONST.MODAL.MODAL_TYPE.CENTERED_UNSWIPEABLE,
             isConfirmButtonDisabled: false,
             confirmButtonFadeAnimation: new Animated.Value(1),
+            sourceURL: props.isAuthTokenRequired ? addEncryptedAuthTokenToURL(props.sourceURL) : props.sourceURL,
         };
 
         this.submitAndClose = this.submitAndClose.bind(this);
         this.closeConfirmModal = this.closeConfirmModal.bind(this);
+        this.onArrowPress = this.onArrowPress.bind(this);
         this.validateAndDisplayFileToUpload = this.validateAndDisplayFileToUpload.bind(this);
         this.updateConfirmButtonVisibility = this.updateConfirmButtonVisibility.bind(this);
     }
 
+    // this prevents a bug in iOS that would show the last image before closing then opening on a new image
+    static getDerivedStateFromProps(props, state) {
+        if (state.isModalOpen && props.sourceURL && state.isModalOpen !== state.prevIsModalOpen) {
+            return {
+                prevIsModalOpen: true,
+                file: {name: lodashGet(props, 'originalFileName', '')},
+                sourceURL: props.isAuthTokenRequired ? addEncryptedAuthTokenToURL(props.sourceURL) : props.sourceURL,
+            };
+        }
+
+        if (!state.isModalOpen && state.isModalOpen !== state.prevIsModalOpen) {
+            return {prevIsModalOpen: false};
+        }
+
+        return null;
+    }
```

Comment:
> I believe in iOS the modal class doesn't get removed in memory and the data here doesn't get updated if I chose another image and try to load it. This updates the image accordingly because it's called at a specific time that reloads the data correctly.

Then we should investigate why & find where the problem lies and fix it, which modal class you are referring to?

Reply:
I retested the app without this on iOS and it works as expected now. I'm able to get rid of it!